cmd/viam-agent/main_windows.go

```diff
@@ -59,23 +69,16 @@ func main() {
 	}()
 
 	goutils.UncheckedError(elog.Info(1, fmt.Sprintf("starting %s service", serviceName)))
-	go commonMain()
-	// note: svc.Run hangs until windows terminates the service. Then we manually call
-	// globalCancel, which stops the go commonMain goroutine, then we wait for the waitgroup.
-	err = svc.Run(serviceName, &agentService{})
-	if err != nil {
-		goutils.UncheckedError(elog.Error(1, fmt.Sprintf("%s service failed: %v", serviceName, err)))
-		return
-	}
-	if globalCancel == nil {
-		goutils.UncheckedError(elog.Error(1, "globalCancel is nil, shutdown will be unclean"))
-	} else {
-		globalCancel()
-	}
-	// wait first so viam-server doesn't try to restart
-	activeBackgroundWorkers.Wait()
-	// KillTree to catch any stragglers
-	if err := utils.KillTree(-1); err != nil {
+
+	go func() {
+		// note: svc.Run() hangs until windows terminates the service via Execute()
+		if err := svc.Run(serviceName, &agentService{}); err != nil {
+			goutils.UncheckedError(elog.Error(1, fmt.Sprintf("%s service failed: %v", serviceName, err)))
+		}
+	}()
+
+	commonMain()
+	if err := zapChildren(); err != nil {
```

Comment:
I don't know if this is technically needed or not. It was the original utils.KillTree (but it doesn't kill the tree, just the leaves) and seems to only be to try to collect/kill orphans, which should be cleaned up by viam-server itself now that we're doing things gracefully. But... extra clean-up didn't seem to do any harm, and if it helps fence in bad modules code, why not.

Also,  I ALMOST renamed it OrphanCrushingMachine() :-)